Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js

```diff
@@ -0,0 +1,319 @@
+            return;
+
+        this.#values.viewport = viewport;
+        this.#updateViewportRow();
+    }
+
+    #setDevicePixelRatio(devicePixelRatio)
+    {
+        if (this.#values.devicePixelRatio === devicePixelRatio)
+            return;
+
+        this.#values.devicePixelRatio = devicePixelRatio;
+        this.needsLayout();
+    }
+
+    #updateViewportRow()
+    {
+        this.#ui.viewportRow.pendingValue = WI.UIString("%dx%d (%dx)").format(this.#values.viewport?.width ?? 0, this.#values.viewport?.height ?? 0, this.#values.devicePixelRatio ?? 1);
+        this.needsLayout();
+    }
+
+    #setQuality(quality)
+    {
+        if (Object.shallowEqual(this.#values.quality, quality))
+            return;
+
+        this.#values.quality = quality;
+        let formatString = WI.UIString("%d dropped of %d", "Dropped Frame Format @ Media Sidebar", "Format string for Dropped Frame count in Media Sidebar");
+        this.#ui.framesRow.pendingValue = formatString.format(quality?.droppedVideoFrames ?? 0, quality?.totalVideoFrames ?? 0);
+        this.needsLayout();
+    }
+
+    #setVideoFormat(videoFormat)
+    {
+        if (this.#values.videoFormat === videoFormat)
+            return;
+
+        this.#values.videoFormat = videoFormat;
+        this.#ui.videoFormatRow.pendingValue = videoFormat;
+        this.needsLayout();
+    }
+
+    #setAudioFormat(audioFormat)
+    {
+        if (this.#values.audioFormat === audioFormat)
+            return;
+
+        this.#values.audioFormat = audioFormat;
+        this.#ui.audioFormatRow.pendingValue = audioFormat;
+        this.needsLayout();
+    }
+
+    #setVideo(video)
+    {
+        if (this.#values.video === video)
+            return;
+
+        if (this.#values.video?.width === video?.width
+            && this.#values.video?.height === video?.height
+            && this.#values.video?.framerate === video?.framerate
+            && this.#values.video?.codec === video?.codec
+            && Object.shallowEqual(this.#values.video?.colorSpace, video?.colorSpace)
+            && this.#values.video?.colorSpace?.primaries === video?.colorSpace?.primaries
+            && this.#values.video?.colorSpace?.matrix === video?.colorSpace?.matrix
+            && this.#values.video?.fullRange === video?.fullRange)
+            return;
+
+        this.#values.video = video;
+        this.#ui.videoSection.element.classList.toggle("hidden", !video);
+        this.#ui.resolutionRow.pendingValue = WI.UIString("%dx%d (%dfps)").format(video?.width ?? 0, video?.height ?? 0, video?.framerate ?? 0);
+        this.#ui.resolutionRow.element.classList.toggle("hidden", !video);
+        this.#ui.videoCodecRow.pendingValue = video?.codec;
+        this.#ui.transferRow.pendingValue = video?.colorSpace?.transfer;
+        this.#ui.primariesRow.pendingValue = video?.colorSpace?.primaries;
+        this.#ui.matrixRow.pendingValue = video?.colorSpace?.matrix;
```

Comment:
No, because then we'd need `} else {` blocks for each, so they clear out the values when going from having `video` to not, and etc. This is much cleaner.